src/meshes/p4est_mesh.jl

```diff
@@ -497,45 +497,319 @@ function p4est_connectivity_from_hohqmesh_abaqus(meshfile, initial_refinement_le
+    # Support only basic quads, membranes, and shells here.
+    quadratic_quads = r"^(CPE8|CPS8|CAX8|S8|M3D8|M3D9).*$"
+
+    # In 3D only standard hexahedra/bricks are supported.
+    linear_hexes = r"^(C3D8).*$"
+    quadratic_hexes = r"^(C3D27).*$"
+
+    # Preprocess the meshfile to remove lower-dimensional elements
+    meshfile_preproc, elements_begin_idx, sets_begin_idx = preprocess_standard_abaqus(meshfile,
+                                                                                      linear_quads,
+                                                                                      linear_hexes,
+                                                                                      quadratic_quads,
+                                                                                      quadratic_hexes,
+                                                                                      n_dimensions)
+
+    # Copy of mesh for p4est with linear elements only
+    meshfile_p4est_rdy, mesh_polydeg = preprocess_standard_abaqus_for_p4est(meshfile_preproc,
+                                                                            linear_quads,
+                                                                            linear_hexes,
+                                                                            quadratic_quads,
+                                                                            quadratic_hexes,
+                                                                            elements_begin_idx,
+                                                                            sets_begin_idx)
+
     # Create the mesh connectivity using `p4est`
-    connectivity = read_inp_p4est(meshfile, Val(n_dimensions))
+    connectivity = read_inp_p4est(meshfile_p4est_rdy, Val(n_dimensions))
     connectivity_pw = PointerWrapper(connectivity)
 
     # These need to be of the type Int for unsafe_wrap below to work
-    n_trees::Int = connectivity_pw.num_trees[]
+    n_trees::Int = connectivity_pw.num_trees[] # = number elements
     n_vertices::Int = connectivity_pw.num_vertices[]
 
+    # Extract a copy of the element vertices to compute the tree node coordinates
+    # `vertices` store coordinates of all three dimensions (even for the 2D case)
+    # since the Abaqus `.inp` format always stores 3D coordinates.
     vertices = unsafe_wrap(Array, connectivity_pw.vertices, (3, n_vertices))
+
     tree_to_vertex = unsafe_wrap(Array, connectivity_pw.tree_to_vertex,
                                  (2^n_dimensions, n_trees))
 
     basis = LobattoLegendreBasis(RealT, polydeg)
     nodes = basis.nodes
 
+    if mesh_polydeg == 2
+        mesh_nnodes = mesh_polydeg + 1 # = 3
+        # Note: We ASSUME that the additional node between the end-vertices lies 
+        # on the center on that line, such that we can use Chebyshev-Gauss-Lobatto nodes!
+        # For polydeg = 2, we have the 3 nodes [-1, 0, 1] (within the reference element).
```

Comment:
Interesting...so in the future for element boundaries higher than quadratic would this become `mesh_polydeg >= 2`? 

This comment is also hyper-specific for the `polydeg=2` case. In general the CGL nodes are nonuniform in `[-1,1]`, include the endpoints, and will only include the point `x=0` for even `polydeg` (i.e. an odd number of points).

Reply:
Yeah so this comment is heavily tailored towards the quadratic case, which is as far as I now the highest degree supported by standard Abaqus (see also my answer above).

Reply:
Okay, no worries. Since the ABAQUS format is only designed to handle up quadratic boundaries, such specific code is fine.

Reply:
I guess it makes sense to add such a comment also here to explain this code.